appinventor/buildserver/src/com/google/appinventor/buildserver/ProjectBuilder.java

```diff
@@ -271,13 +271,17 @@ private static Map<String, String> getScreenOrientations(List<String> files)
     return result;
   }
 
-  private static Set<String> getComponentTypes(List<String> files, File assetsDir)
+  private static Set<String> getComponentTypes(List<String> files, File assetsDir, File projectRoot)
```

Comment:
Should we apply this logic to checking the bky files as well?

Reply:
Probably yes. I have now refactored the change and made it a bit more generic: now `sourceFiles` is already filtered, and contains only the source files in `src/`. I have checked where that list was being used, and it was for component types, properties, blocks and orientations. I assume all of them should require to only process those files from the `src/` folder.

Also, found some unused constants to define each type of file, so hopefully now it's more "consistent"...